src/hooks/useOrgHooks.js

```diff
@@ -201,44 +167,55 @@ export function useVaultBalance(installedApps, token, timeout = 1000) {
   return vaultBalance
 }
 
-export function useTokenBalances(account, token) {
+export function useTokenBalances(account, token, timer = 3000) {
   const [balances, setBalances] = useState({
     balance: new BigNumber(-1),
     totalSupply: new BigNumber(-1),
   })
 
+  const tokenContract = useContractReadOnly(token.id, minimeTokenAbi)
+
   useEffect(() => {
     if (!token.id) {
       return
     }
 
     let cancelled = false
+    let timeoutId
 
-    const queryAccountStakeBalance = async () => {
-      const wrapper = new GraphQLWrapper(TOKEN_MANAGER_SUBGRAPH)
-      const results = await wrapper.performQuery(gql(TOKEN_HOLDER_QUERY), {
-        id: `tokenAddress-${token.id}`,
-        address: account || '',
-      })
-
-      if (!results.data) {
-        return
-      }
+    const fetchAccountStakeBalance = async () => {
+      try {
+        let contractNewBalance = new BigNumber(-1)
+        if (account) {
+          contractNewBalance = await tokenContract.balanceOf(account)
+        }
 
-      const { miniMeToken } = results.data
+        const contractTotalSupply = await tokenContract.totalSupply()
 
-      if (!cancelled) {
-        setBalances({
-          balance: new BigNumber(miniMeToken.holders[0]?.balance || 0),
-          totalSupply: new BigNumber(miniMeToken.totalSupply),
-        })
+        if (!cancelled) {
+          // Contract value is bn.js so we need to convert it to bignumber.js
```

Comment:
Wondering why we're using bignumber.js here? Is it for the decimal utils?

Reply:
yeah, for example the total supply is used to do the maths to calculate the convction and threshold and since they are really big numbers  with lot of decimals we need it https://github.com/1Hive/honey-pot/blob/master/src/hooks/useProposals.js#L50-L62

Reply:
Aaah yeah, I meant why we're using `bignumber.js` rather than `bn.js` 😛

Reply:
ahh because it alow us to use .xxxxxxxxxxxx really big with bn.js was not accurate at all

Reply:
Ah gotcha! Makes sense. I was mainly saying it because the native BigInts that come on Chrome and Firefox (only Safari's behind :/ ) [only support integers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt), so I generally prefer `bn.js` or `JSBI` which is actually the closest to the native BigInt standard. That said though, having decimal support is super nice!